sdk/Pulumi/Provider/Provider.cs

```diff
@@ -611,6 +615,9 @@ class ResourceProviderService : ResourceProvider.ResourceProviderBase, IDisposab
         private Provider? implementation;
         private readonly string version;
         private string? engineAddress;
+        // More recent versions of the engine send URN and ID of the provider to `Configure`. With that we can construct
+        // a `DependencyProviderResource` to use in `Construct` and `Call` to refer to the provider itself.
+        private string? providerSelfReference;
```

Comment:
There's no value in sending a ProviderResource to any of the normal CRUD methods, but maybe we should send the URN for logging reasons? That would allow you to log a message to the provider resource, while performing a Create/Update for another resource.